File: Data Structures & Algorithms/number-of-good-pairs/submission-0.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        res = 0
        m = {}
        for j in range(len(nums)):
            if nums[j] in m:
                res += m[nums[j]]
            else:
                m[nums[j]] = 0
            m[nums[j]] += 1

        return res
        

File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        int n = nums.length;
        Set set = new HashSet<Integer>();
        for (int num : nums) {
            if (!set.add(num)) {
                return true;
            }
        }
        return false;
    }
}